code/datums/materials/meat.dm

```diff
@@ -31,3 +31,23 @@
 	var/oil_count = 2 * (amount / MINERAL_MATERIAL_AMOUNT)
 	source.AddComponent(/datum/component/edible, list(/datum/reagent/consumable/nutriment = nutriment_count, /datum/reagent/consumable/cooking_oil = oil_count), null, RAW | MEAT | GROSS, null, 30, list("Fleshy"))
 
+
+/datum/material/meat/mob
+	init_flags = MATERIAL_INIT_BESPOKE
+
+/datum/material/meat/mob/Initialize(_id, mob/living/source)
+	if(!istype(source))
+		return FALSE
+
+	name = "[source?.name ? "[source.name]'s" : "mystery"] [initial(name)]"
+	return ..()
+
```

Comment:
I don't like using keywords in typepaths like this, could you pick something else?

Reply:
Sure, what would you recommend?

Reply:
Maybe named? Honestly not too sure what would work well here